src/lib/connectors/shared/remote-debugging-connector.ts

```diff
@@ -326,7 +326,12 @@ export class Connector implements IConnector {
             };
         }
 
-        if (hasAttributeWithValue(data.element, 'link', 'rel', 'icon')) {
+        const hasFaviconLinkTag = hasAttributeWithValue(data.element, 'link', 'rel', 'icon');
+        // If favicon is not present in the html, CDP will send a request to get favicon from the root directory.
+        // We should set the tag to true in this case to avoid fetching favicon twice.
+        const hasFaviconInRootDir = data.request.url.endsWith('/favicon.ico');
```

Comment:
@molant I think your comment is correct. There is a scenario that `/images/favicon.ico` is included in the html as the resource a `img` instead of a `link` tag in the head (`rel` set as `icon`). In this scenario we shouldn't set the flag to true since the favicon image isn't included in the page in an appropriate way. We should still go into `getFavicon` trying to manually download the favicon.